subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/filter/TestSelectionMatcher.java

```diff
@@ -25,13 +25,21 @@
 
 public class TestSelectionMatcher {
 
-    private final List<Pattern> includePatterns;
+    private final List<Pattern> buildScriptIncludePatterns;
+    private final List<Pattern> commandLineIncludePatterns;
 
-    public TestSelectionMatcher(Collection<String> includedTests) {
-        includePatterns = new ArrayList<Pattern>(includedTests.size());
+    public TestSelectionMatcher(Collection<String> includedTests, Collection<String> includedTestsCommandLine) {
+        buildScriptIncludePatterns = preparePatternList(includedTests);
+        commandLineIncludePatterns = preparePatternList(includedTestsCommandLine);
+    }
+
+    private List<Pattern> preparePatternList(Collection<String> includedTests) {
+        List<Pattern> buildScriptIncludePatterns;
+        buildScriptIncludePatterns = new ArrayList<Pattern>(includedTests.size());
```

Comment:
Can't these two lines merge into one line? And is this local variable name `buildScriptIncludePatterns` appropriate when it can represent both `buildScriptIncludePatterns` and `commandLineIncludePatterns` in the context.